### PR TITLE
DE-3241: athena checker fix

### PR DESCRIPTION
**Pull request by setiat**
Partition path in Athena.json can now be root level property from IDS schema JSON

Removed  definition and usage of `AthenaChecker.check_athena_path_and_root_properties_conflict`. Updated unit tests

### CODE REVIEW

**Review by wkopen — CHANGES_REQUESTED** [review]
The code change looks good. But can we add a test for the happy path to test athena path using the top-level field will not fail the validator?

**Review by wkopen — CHANGES_REQUESTED** [review]
In unit test, you should only add what you need for the test. In this case, you are adding the entire schema but you are only testing the top-level fields. The rest of the code will just be maintenance burden. Please remove what you don't need.

**Comment by setiat** [comment]
> In unit test, you should only add what you need for the test. In this case, you are adding the entire schema but you are only testing the top-level fields. The rest of the code will just be maintenance burden. Please remove what you don't need.

minimised data all Athena test cases

File: __tests__/unit/generic/files/athena/valid_root_level_property_partition/athena.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+    "root": "root",
+    "partitions": [
+        {
+            "path": "root_level_property",
+            "name": "root_level_property_partition"
+        },
+        {
+            "path": "root_level_property.nested_property",
```
Comment (wkopen) [inline __tests__/unit/generic/files/athena/valid_root_level_property_partition/athena.json:9]: According to the rule defined in https://developers.tetrascience.com/docs/ids-design-conventions-athenajson#partition-path, you can't have a nested property in the path. So it's an invalid case. The validator should fail.

**Review by wkopen — CHANGES_REQUESTED** [review]
Thanks for trimming the test case down! Now you've introduced another violation of the rule. Please see my inline comment.

File: __tests__/unit/generic/files/athena/valid_root_level_property_partition/athena.json (hunk 1 above)
Comment (setiat) [reply __tests__/unit/generic/files/athena/valid_root_level_property_partition/athena.json]: @wkopen,
`root_level_property` is of `object` type and `nested_property` has `string` type. So `"root_level_property.nested_property"` should be a valid Path.

I have renamed `nested_property` to `child_property` as former was misleading.

Comment (wkopen) [reply __tests__/unit/generic/files/athena/valid_root_level_property_partition/athena.json]: thanks! `nested` is a ES keyword so I guess that's where the confusion is from

**Pull request merged**